### ISSUE TITLE
Inconsistent run when using pre-trained h5 weight file on Theano vs TensorFlow/CNTK

### ISSUE THREAD

**Issue by hvpham** [keras-team/keras#10943]
Hi,
I have a pre-trained model that I try to use. It has an h5 weight file that I loaded into Keras.
But when I run it using Theano, the output is not the same as TensorFlow/CNTK (TensorFlow and CNTK agree with each other)

On further investigation, I found out that the attribute original_backend in this file is set to "plaidml" which from what I understand is one of the computational engines for TensorFlow. But in the function "_need_convert_kernel" in engine/saving.py, it does not convert if the original_backend is not in the lookup table. Which mean when I sue Theano we need to convert the weight but Keras does not which lead to inconsistent output. I have top1 accuracy for TensorFlow and CNTK around 91% but only 44% for Theano.
